### PR TITLE
Fix SonarCloud Warning for area package

### LINKED ISSUE #578: Clean up/Refactor areas Package
# **Description**
**Task:**
Refactor the areas package to improve modularity and coupling and reduce code smells and risks of bugs.
 
# **Milestones**
List of steps that need to be completed for this task.
- [x] Refactor config files to not use json (Aug 9)
- [x] Refactor areas to separate enums (Aug 9)
- [x] Reduce Code Smells (Aug 10)
- [x] Reduce Code Duplication (Aug 11)
 
**Completion Deadline:** Aug. 11
 
# **Member**
- Mittun Sudhahar (@mitsudz )
- Howie (@Howard-12)

### PR DESCRIPTION
**Pull request by Howard-12**
# Description

Clean up code to meet SonarCloud requirements and refactor code when necessary. Functionality remains the same. 
Closes #578 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?
The game run without crashes, and terrains are generated as expected : )

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: source/core/src/main/com/csse3200/game/areas/MapHandler.java [hunk 1]
```diff
@@ -93,32 +85,55 @@ public static void unlockNextArea() {
    * @return true iff the map is unlocked
    */
   public static boolean getUnlockedOcean() {
```
Comment (mitsudz) [inline source/core/src/main/com/csse3200/game/areas/MapHandler.java:87]: Can some of these functions just be combined and take in an enum, e.g. getUnlocked(MapType type)?

File: source/core/src/main/com/csse3200/game/areas/terrain/TerrainChunk.java [hunk 2]
```diff
@@ -55,10 +55,10 @@ public void generateTiles(GridPoint2 chunkPos, Map<GridPoint2, TerrainChunk> loa
     // INFO: The Map is equally divied into three areaas. Each area is 16x10 tiles wide.
     inArea = checkAreaType(position);
     switch (inArea) {
-      case MapType.FOREST -> totalTiles = TerrainResource.getTileSize(TileLocation.FOREST);
-      case MapType.WATER -> totalTiles = TerrainResource.getTileSize(TileLocation.WATER);
-      case MapType.FOG -> totalTiles = TerrainResource.getTileSize(TileLocation.FOG);
-      default -> totalTiles = TerrainResource.getTileSize(TileLocation.AIR);
+      case MapType.FOREST -> totalTiles = terrainResource.getForestSize();
```
Comment (mitsudz) [inline source/core/src/main/com/csse3200/game/areas/terrain/TerrainChunk.java:58]: I'm not sure this change is an improvement - it requires a function for each map type which is pointless considering they do the same thing.

File: source/core/src/main/com/csse3200/game/areas/terrain/TerrainResource.java [hunk 3]
```diff
@@ -92,6 +93,22 @@ public TerrainResource(MapHandler.MapType mapType) {
         this.setPossibleTiles();
     }
 
+    public int getForestSize() {
```
Comment (mitsudz) [inline source/core/src/main/com/csse3200/game/areas/terrain/TerrainResource.java:96]: I think it's reasonable to combine these four functions to just take in an enum for the tile/map type

**Review by mitsudz — CHANGES_REQUESTED** [review]
A few small styling changes requested.

**Review by mitsudz — APPROVED** [review]
Looks good!

**Review by tompreston1 — APPROVED** [review]

**Review by PHONGNGYUEN1 — APPROVED** [review]
Looks pretty good. approved

**Pull request merged**